doc/release-versioning.md

```diff
@@ -0,0 +1,151 @@
+
+Milestone labels have the form: `milestone-x.y.0`, where `x` and `y` are major and minor SDK versions, respectively. This particular milestone demarcates the SDK `x.y.0` release; once issues within this milestone are addressed, the release process can begin.
+
+## Major versions
+
+Major version changes can break compatibility between the previous major versions; they are not necessarily backwards or forwards compatible. SDK change targets include but are not limited to:
+- `operator-sdk` command and sub-commands
+- Golang API
+- Formats of various yaml manifest files
+
+## Minor versions
+
+Minor version changes will not break compatibility between the previous minor versions; to do so is a bug. SDK changes will involve addition of optional features, non-breaking enhancements, and *minor* bug fixes identified from previous versions.
+
+### Creating a minor version branch
+
+We expect to release patches for minor releases, so we create a patch trunk to branch from. The naming convention follows "v2.1.x", where the major version is 2, minor is 1, and "x" is a patch version placeholder.
+
+```bash
+$ git checkout \
+    -b "v${MAJOR_VERSION}.${NEW_MINOR_VERSION}.x" \
+    tags/"v${MAJOR_VERSION}.${NEW_MINOR_VERSION}.0"
+$ git push git@github.com:operator-framework/operator-sdk.git \
+    "v${MAJOR_VERSION}.${NEW_MINOR_VERSION}.x"
+```
+
+## Patch versions
+
+Patch versions changes are meant only for bug fixes, and will not break compatibility of the current minor version. A patch release will contain a collection of minor bug fixes, or individual major and security bug fixes, depending on severity.
+
+### Creating a patch version branch
+
+As more than one patch may be created per minor release, patch branch names of the form "${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION}" will be created after a bug fix has been pushed, and the bug fix branch merged into the patch branch only after testing.
+
+```bash
+$ git checkout "v${MAJOR_VERSION}.${MINOR_VERSION}.x"
+$ git checkout -b "cherry-picked-change"
+$ git cherry-pick "$GIT_COMMIT_HASH"
+$ git push origin "cherry-picked-change"
+```
+
+# Releases
+
+Making an Operator SDK release involves:
+- Tagging and signing a git commit and pushing the tag to GitHub.
+- Building a release binary, signing the binary, and uploading both binary and signature to GitHub.
+
+Releases can only be performed by [maintainers][doc-maintainers].
+
+## Binaries
+
+Release binaries will be built for the `x86_64` architecture. Binaries will be signed using a maintainers' [PGP key][doc-maintainer-pgp-keys], and signatures will be uploaded to the release along with its accompanying tarball-ed binary. Ensure you import maintainer keys to verify release binaries. Note: `gpg` and `gpg2` can be used interchangeably to sign and verify binaries; substitute `gpg2` for `gpg` below.
+
+Creating a release binary tarball:
+```bash
+$ make build
```

Comment:
`make build` just builds the binary for architecture and os of the current machine. We should have a `make release` function that cross-compiles for all supported platforms (linux and macos x86_64). I built a release for ct last year, which has a pretty well designed release process. This is their makefile (they support many more platforms than us, but still a good reference): https://github.com/coreos/container-linux-config-transpiler/blob/master/Makefile

Reply:
Ah excellent. I wasn't sure which architectures were supported so I assumed only linux x86_64 was. I will make updates to the Makefile.